pkg/plugin/plugin_v2.go

```diff
@@ -183,12 +184,12 @@ func (p *V2Plugin) Decrypt(ctx context.Context, request *pb.DecryptRequest) (*pb
 
 	result, err := p.svc.Decrypt(ctx, input)
 	if err != nil {
-		select {
-		case p.healthCheck.healthCheckErrc <- err:
-		default:
+		errorType := kmsplugin.ParseError(err).String()
+		if errorType != kmsplugin.KMSErrorTypeCorruption.String() {
```

Comment:
is it possible to add unit test for this use case?

Reply:
I added logic to the TestHealthV2 test to cover this